src/brogue/Items.c

```diff
@@ -3460,14 +3460,15 @@ void getImpactLoc(pos *returnLoc, const pos originLoc, const pos targetLoc,
     pos coords[DCOLS + 1];
     short i, n;
     creature *monst;
+    creature *orig = monsterAtLoc(originLoc.x, originLoc.y);
 
     n = getLineCoordinates(coords, originLoc, targetLoc, theBolt);
     n = min(n, maxDistance);
     for (i=0; i<n; i++) {
         monst = monsterAtLoc(coords[i].x, coords[i].y);
-        if (monst
-            && !monsterIsHidden(monst, monsterAtLoc(originLoc.x, originLoc.y))
+        if (canAttack(orig, monst)
```

Comment:
This is wrong from the POV of getImpactLoc. Bolts do not pass through creatures if the caster is unable to attack them.